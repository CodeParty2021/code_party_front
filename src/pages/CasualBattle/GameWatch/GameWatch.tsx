import React from "react";
import { useGameWatchState } from "./hooks/useGameWatchState";

type Props = {};

export const CasualBattleGameWatch: React.FC<Props> = () => {
  const {isAnalysing, exitBtnHandler} = useGameWatchState();
  return (
    <div>
      <div>結果</div>
      <div>
        {isAnalysing ? "解析中" : "観戦中"}
      </div>
      <button onClick={exitBtnHandler}>退出</button>
    </div>
  );
};
