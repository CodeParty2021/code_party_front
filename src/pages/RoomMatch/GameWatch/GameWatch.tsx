import React from "react";
import { useGameWatchState } from "./hooks/useGameWatchState";
import { useRunSimulation } from "./hooks/useRunSimulation";
import { Background } from "pages/Code/Coding/CodingStyle";
import { BackLink } from "components/BackLink/BackLink";
import {
  GameWatchStyle,
  messageProps,
  UnityStyle,
  WatchingLogo,
} from "./GameWatchStyle";
import { Message } from "components/Message/Message";
import { AlgoHead } from "components/Character/Algo/Head/AlgoHead";

type Props = {};

export const RoomMatchGameWatch: React.FC<Props> = () => {
  useRunSimulation();
  const { unityContext, exitBtnHandler, state, messageType } =
    useGameWatchState();

  return (
    <GameWatchStyle state={state}>
      <Background color="blue" />
      <div className="gamewatch_unity">
        <UnityStyle unityContext={unityContext} />
        <WatchingLogo src="/img/watching_logo.svg" wrapper="svg" />
      </div>
      <BackLink
        iconColor="blue"
        backMessage="待機部屋に戻る"
        onClick={exitBtnHandler}
      />
      <div className="gamewatch_message">
        <Message {...messageProps(messageType)} />
        <AlgoHead width="142px" height="117px" color="blue" />
      </div>
    </GameWatchStyle>
  );
};
