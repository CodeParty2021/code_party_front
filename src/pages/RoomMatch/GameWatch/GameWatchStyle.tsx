import { Message } from "components/Message/Message";
import { ComponentProps } from "react";
import { ReactSVG } from "react-svg";
import styled, { css } from "styled-components";
import { WHITE } from "styles/colors";
import { IResponse } from "./hooks/useGameWatchState";
import { BackLinkButton } from "components/BackLink/BackLinkStyle";
import Unity from "react-unity-webgl";
import { FlexGap } from "styles/FlexGap/FlexGap";

type PrivateProps = {
  state: IResponse["state"];
};

const defaultStyle = css`
  position: relative;
  width: 100vw;
  height: 100vh;

  /* BackLinkStyle */
  ${BackLinkButton} {
    position: absolute;
    top: 32px;
    left: 32px;
  }

  .gamewatch_message {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    ${FlexGap({ gap: "8px", direction: "column" })}

    position: absolute;
    top: 0;
    left: 50%;
    transform: translate(-50%, -100%);
    transition: all 0.5s ease;
  }

  .gamewatch_unity {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 12px 48px 0;
    ${FlexGap({ gap: "8px", direction: "column" })}
    transition: opacity 0.5s ease;
  }
`;

const analyzingStyle = css`
  .gamewatch_message {
    top: 50%;
    transform: translate(-50%, calc(-50% - 110px));
  }

  .gamewatch_unity {
    opacity: 0;
  }
`;

const runningStyle = css``;

const finishStyle = css``;

export const GameWatchStyle = styled.div<PrivateProps>`
  ${defaultStyle}

  ${({ state }) => state === "Analyzing" && analyzingStyle}
  ${({ state }) => state === "Running" && runningStyle}
  ${({ state }) => state === "Finish" && finishStyle}
  ${({ state }) => state === "AnalyzingError" && analyzingStyle}
`;

export const UnityStyle = styled(Unity)`
  width: calc(90vh * 4 / 3);
  height: 90vh;
  border-radius: 12px;
  flex-grow: 0;
  flex-shrink: 0;
`;

export const WatchingLogo = styled(ReactSVG)`
  width: 50px;
  height: 32px;
  fill: ${WHITE};
`;

export const messageProps = (
  state: IResponse["state"]
): ComponentProps<typeof Message> => {
  if (state === "Analyzing")
    return {
      title: "シミュレーションチュウ",
      value: "しばらくお待ちください",
      color: "blue",
    };
  else if (state === "AnalyzingError")
    return {
      title: "エラー",
      value: "左上の戻るボタンをおしてください！",
      color: "red",
    };
  else
    return {
      title: "このメッセージは",
      value: "出ないはずだよ",
      color: "blue",
    };
};
