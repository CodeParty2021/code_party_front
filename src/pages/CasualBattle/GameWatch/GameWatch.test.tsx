import React from "react";
import { shallow } from "enzyme";
import { CasualBattleGameWatch } from "./GameWatch";

describe("<CasualBattleLobby />", () => {
  it("auth snapshot test", () => {
    const wrapper = shallow(<CasualBattleGameWatch />);

    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
