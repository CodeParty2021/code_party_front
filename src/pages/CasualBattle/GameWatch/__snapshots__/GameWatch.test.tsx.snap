// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CasualBattleLobby /> auth snapshot test 1`] = `
Array [
  <div>
    <div>
      結果
    </div>
    <div>
      aaa
    </div>
    <div>
      <Link
        to="/casual-battle/waiting-room"
      >
        ロビーに戻る
      </Link>
    </div>
  </div>,
]
`;
